src/System.CommandLine.Hosting.Tests/HostingHandlerTest.cs

```diff
@@ -0,0 +1,183 @@
+using System.CommandLine.Invocation;
+using System.CommandLine.IO;
+using System.CommandLine.Parsing;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentAssertions;
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+
+namespace System.CommandLine.Hosting.Tests
+{
+    public static class HostingHandlerTest
+    {
+
+        [Fact]
+        public static void Constructor_Injection_Injects_Service()
+        {
+            var service = new MyService();
+
+            var parser = new CommandLineBuilder(
+                new MyCommand()
+                )
+                .UseHost((builder) => {
+                    builder.ConfigureServices(services =>
+                    {
+                        services.AddTransient(x => service);
+                    })
+                    .UseCommandHandler<MyCommand, MyCommand.MyHandler>();
+                })
+                .Build();
+
+            var result = parser.InvokeAsync(new string[] { "--int-option", "54"})
+                .GetAwaiter().GetResult();
+
+            service.Value.Should().Be(54);
+        }
+
+        [Fact]
+        public static void Parameter_is_available_in_property()
+        {
+            var parser = new CommandLineBuilder(new MyCommand())
+                .UseHost(host =>
+                {
+                    host.ConfigureServices(services =>
+                    {
+                        services.AddTransient<MyService>();
+                    })
+                    .UseCommandHandler<MyCommand, MyCommand.MyHandler>();
+                })
+                .Build();
+
+            var result = parser.InvokeAsync(new string[] { "--int-option", "54"})
+                .GetAwaiter().GetResult();
+
+            result.Should().Be(54);
+        }
+
+        [Fact]
+        public static void Can_have_diferent_handlers_based_on_command()
+        {
+            var root = new RootCommand();
+
+            root.AddCommand(new MyCommand());
+            root.AddCommand(new MyOtherCommand());
+            var parser = new CommandLineBuilder(root)
+                .UseHost(host =>
+                {
+                    host.ConfigureServices(services =>
+                    {
+                        services.AddTransient<MyService>();
+                    })
+                    .UseCommandHandler<MyCommand, MyCommand.MyHandler>()
+                    .UseCommandHandler<MyOtherCommand, MyOtherCommand.MyHandler>();
+                })
+                .Build();
+
+            var result = parser.InvokeAsync(new string[] { "mycommand", "--int-option", "54" })
+                .GetAwaiter().GetResult();
+
+            result.Should().Be(54);
+
+            result = parser.InvokeAsync(new string[] { "myothercommand", "--int-option", "54" })
+                .GetAwaiter().GetResult();
+
+            result.Should().Be(540);
```

Comment:
Can we embed the code that produces the result into the test? Because the handler implementation has this piece of trivia (multiplying the parsed `int` by 10), it means the test doesn't describe the behavior as clearly as it might.